docs/concepts/controllers.md

```diff
@@ -39,42 +38,46 @@ Once the manager is `Start()`ed, all the controllers that are *registered* with
 
-- Condition: The controller updates the `Conditions` field which holds the latest information of the `Etcd`'s state. The condition checks that are performed are `AllMembersCheck`, `ReadyCheck` and `BackupReadyCheck`. The first two of these checks make use of the `EtcdMemberStatus` to update `Conditions` with information about the members, and the last corresponds to the status of the backup.
+- `LastOperation` holds information about the last operation performed on the etcd cluster, indicated by fields `Type`, `State`, `Description` and `LastUpdateTime`. Additionally, a field `RunID` indicates the unique ID assigned to the specific reconciliation run, to allow for better debugging of issues.
+- `LastErrors` is a slice of errors encountered by the last reconciliation run. Each error consists of fields `Code` to indicate the custom druid error code for the error, a human-readable `Description`, and the `ObservedAt` time when the error was seen.
+- `ObservedGeneration` indicates the latest `generation` of the `Etcd` resource that druid has "observed" and consequently reconciled. It helps identify whether a change in the `Etcd` resource spec was acted upon by druid or not.
 
-To reflect changes that occur in the `Statefulset` status in the `Etcd` resource, the *custodian controller* keeps a watch on the `Statefulset`.
+Status fields of the `Etcd` resource which correspond to the `StatefulSet` like `CurrentReplicas`, `ReadyReplicas` and `Replicas` are updated to reflect those of the `StatefulSet` by the controller.
+
+Status fields related to the etcd cluster itself, such as `Members`, `PeerUrlTLSEnabled` and `Ready` are updated as follows:
+
+- Cluster Membership: The controller updates the information about etcd cluster membership like `Role`, `Status`, `Reason`, `LastTransitionTime` and identifying information like the `Name` and `ID`. For the `Status` field, the member is checked for the *Ready* condition, where the member can be in `Ready`, `NotReady` and `Unknown` statuses.
 
-The *custodian controller* reconciles periodically, which can be set through the `--custodian-sync-period` CLI flag (default being 30 seconds). It also reconciles whenever there are changes to the `Statefulset` status.
+`Etcd` resource conditions are indicated by status field `Conditions`.  The condition checks that are currently performed are:
 
-The *custodian controller* is essential to the functioning of etcd-druid, thus the minimum number of worker threads is 1, (default being 3).
+- `AllMembersReady`: indicates readiness of all members of the etcd cluster.
+- `Ready`: indicates overall readiness of the etcd cluster in serving traffic.
+- `BackupReady`: indicates health of the etcd backups, i.e., whether etcd backups are being taken regularly as per schedule. This condition is applicable only when backups are enabled for the etcd cluster.
```

Comment:
This is an implementation detail. Since we anyway plan to remove `BackupReady` condition in favour of separate `FullSnapshotStale` and `DeltaSnapshotStale` conditions, as described in https://github.com/gardener/etcd-druid/issues/645, we should address this in that PR, not this one, since we have not changed anything wrt to the `BackupReady` condition itself.